src/audio/kpb.c

```diff
@@ -1795,7 +1854,14 @@ static enum task_state kpb_draining_task(void *arg)
 	buffer_release(sink);
 
 	/* Reset host-sink copy mode back to its pre-draining value */
-	sink = buffer_acquire(kpb->host_sink);
+	int sink_idx;
+
+	for (sink_idx = 1; sink_idx < KPB_MAX_NO_OF_SINKS; sink_idx++) {
+		if (draining_data->sink->id == kpb->sinks[sink_idx]->id) {
```

Comment:
in fact when you register clients you do
```
	kpb->clients[cli->id].id  = cli->id;
```
so the ID is always equal to the index. So you don't need to search, you can just use `kpb->sinks[draining_data->sink->id]`. Maybe there are more locations like this, please, check, e.g. in `kpb_copy()`